src/NuGet.Services.AzureSearch/Db2AzureSearch/Db2AzureSearchCommand.cs

```diff
@@ -0,0 +1,228 @@
+            _indexActionBuilder = indexActionBuilder ?? throw new ArgumentNullException(nameof(indexActionBuilder));
+            _serviceClient = serviceClient ?? throw new ArgumentNullException(nameof(serviceClient));
+            _searchIndexClient = searchIndexClient ?? throw new ArgumentNullException(nameof(searchIndexClient));
+            _hijackIndexClient = hijackIndexClient ?? throw new ArgumentNullException(nameof(hijackIndexClient));
+            _options = options ?? throw new ArgumentNullException(nameof(options));
+            _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+        }
+
+        public async Task ExecuteAsync()
+        {
+            var allWork = new ConcurrentBag<NewPackageRegistration>();
+            using (var cancelledCts = new CancellationTokenSource())
+            using (var completedCts = new CancellationTokenSource())
+            {
+                // Initialize the indexes
+                await InitializeAsync();
+
+                // Set up the producer and the consumers.
+                var producerTask = ProduceWorkAsync(allWork, completedCts, cancelledCts.Token);
+                var consumerTasks = Enumerable
+                    .Range(0, _options.Value.WorkerCount)
+                    .Select(i => ConsumeWorkAsync(allWork, completedCts.Token, cancelledCts.Token))
+                    .ToList();
+                var allTasks = new[] { producerTask }.Concat(consumerTasks).ToList();
+
+                // If one of the tasks throws an exception before the work is completed, cancel the work.
+                var firstTask = await Task.WhenAny(allTasks);
+                if (firstTask.IsFaulted)
+                {
+                    cancelledCts.Cancel();
+                }
+
+                await firstTask;
+                await Task.WhenAll(allTasks);
+            }
+        }
+
+        private async Task InitializeAsync()
+        {
+            if (_options.Value.ReplaceIndexes)
+            {
+                await DeleteIndexIfExistsAsync(_options.Value.SearchIndexName);
+                await DeleteIndexIfExistsAsync(_options.Value.HijackIndexName);
+            }
+
+            await CreateIndexAsync<SearchDocument.Full>(_options.Value.SearchIndexName);
+            await CreateIndexAsync<HijackDocument.Full>(_options.Value.HijackIndexName);
+        }
+
+        private async Task DeleteIndexIfExistsAsync(string indexName)
+        {
+            if (await _serviceClient.Indexes.ExistsAsync(indexName))
+            {
+                _logger.LogWarning("Deleting index {IndexName}.", indexName);
+                await _serviceClient.Indexes.DeleteAsync(indexName);
+            }
+        }
+
+        private async Task CreateIndexAsync<T>(string indexName)
+        {
+            _logger.LogInformation("Creating index {IndexName}.", indexName);
+            await _serviceClient.Indexes.CreateAsync(new Index
```

Comment:
what happens if the index name is the same as the existing index?

Reply:
It will throw conflict.